x-pack/metricbeat/module/googlecloud/stackdriver/metricset.go

```diff
@@ -38,19 +43,33 @@ func init() {
 // interface methods except for Fetch.
 type MetricSet struct {
 	mb.BaseMetricSet
-	config config
+	config            config
+	metricsMeta       map[string]metricMeta
+	requester         *stackdriverMetricsRequester
+	StackDriverConfig []stackDriverConfig `config:"metrics" validate:"nonzero,required"`
+}
+
+//stackDriverConfig holds a configuration specific for stackdriver metricset.
+type stackDriverConfig struct {
+	MetricTypes []string `config:"metric_types" validate:"required"`
```

Comment:
these are not actually types but metric names right? how about using `names` here?

Reply:
Hmmmm to be consistent with GCP, it is `metric.type`. `metric.type` is a part of metric selector, which includes its DNS name prefix. Here is an [exmaple](https://cloud.google.com/monitoring/api/v3/filters#overview):
```
metric.type = "compute.googleapis.com/instance/cpu/usage_time" AND
    metric.labels.instance_name = "my-instance-name"
```